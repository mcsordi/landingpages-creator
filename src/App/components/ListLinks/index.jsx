import P from 'prop-types';
import MenuLink from '../MenuLink/index';

function ListLinks({ links }) {
  return (
    <div className="flex flex-col md:flex-row flex-wrap md:max-w-xl">
      {links.map((link, id) => (
        <MenuLink text={link.text} link={link.link} key={id}></MenuLink>
      ))}
    </div>
  );
}
ListLinks.propTypes = {
  links: P.object,
};
export default ListLinks;
