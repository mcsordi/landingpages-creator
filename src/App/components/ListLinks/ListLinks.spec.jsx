import { render, screen } from '@testing-library/react';
import ListLinks from '.';
import links from '../NavBar/mockLinks';

describe('<ListLinks/>', () => {
  it('should render links component', () => {
    render(<ListLinks links={links} />);
    const linkComponent = screen.getAllByRole('link');
    expect(linkComponent).toHaveLength(10);
  });
  it('should render links component with text', () => {
    render(<ListLinks links={links} />);
    const linkComponent = screen.getAllByText(/texto/i);
    expect(linkComponent).toHaveLength(10);
  });
  it('should render links component with links', () => {
    render(<ListLinks links={links} />);
    const linkComponent = screen.getAllByRole('link');
    expect(linkComponent).toHaveLength(10);
    expect(linkComponent[2]).toHaveAttribute('href', '#texto3');
  });
  it('should match a snapshot', () => {
    render(<ListLinks links={links} />);
    const linkComponent = screen.getAllByRole('link');
    expect(linkComponent).toMatchSnapshot();
  });
});
