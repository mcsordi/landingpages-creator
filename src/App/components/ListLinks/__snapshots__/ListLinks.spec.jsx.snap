// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListLinks/> should match a snapshot 1`] = `
Array [
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto1"
    rel="noreferrer"
    target="_self"
  >
    texto1
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto2"
    rel="noreferrer"
    target="_self"
  >
    texto2
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto3"
    rel="noreferrer"
    target="_self"
  >
    texto3
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto4"
    rel="noreferrer"
    target="_self"
  >
    texto4
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto5"
    rel="noreferrer"
    target="_self"
  >
    texto5
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto6"
    rel="noreferrer"
    target="_self"
  >
    texto6
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto7"
    rel="noreferrer"
    target="_self"
  >
    texto7
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto8"
    rel="noreferrer"
    target="_self"
  >
    texto8
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto9"
    rel="noreferrer"
    target="_self"
  >
    texto9
  </a>,
  <a
    class="relative text-primary m-2 py-2 text-xsmall hvr-underline-from-center"
    href="#texto10"
    rel="noreferrer"
    target="_self"
  >
    texto10
  </a>,
]
`;
